mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/MantisSchedulerFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.master.scheduler;
+
+import io.mantisrx.server.master.domain.JobDefinition;
+
+/**
+ * Factory for the Mantis Scheduler based on the JobDefinition
+ */
+public interface MantisSchedulerFactory {
```

Comment:
Shall we keep interface naming with prefix "I"?

Reply:
I'm not too fond of that convention, honestly. I don't mind following it if it's consistently followed. However, from whatever I can see from the current Mantis code, I don't think a lot of components follow that style.